src/ethereum/genesis.py

```diff
@@ -0,0 +1,84 @@
+"""
+Genesis Config Related Functionalities
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+..contents:: Table of Contents
+    :backlinks: none
+    :local:
+
+Introduction
+------------
+
+Functionalities to load the genesis configs and the configurations for
+different chains.
+"""
+import json
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Dict
+
+from ethereum.base_types import U256, Bytes, Bytes8, Uint, slotted_freezable
+from ethereum.frontier.eth_types import Address
+
+from .utils import (
+    hex_to_address,
+    hex_to_bytes,
+    hex_to_bytes8,
+    hex_to_u256,
+    hex_to_uint,
+)
+
+
+@slotted_freezable
+@dataclass
+class GenesisConfig:
+    """
+    Genesis Config has the alloc data for the pre-sale of ether and some
+    fields of the genesis block.
+    """
+
+    difficulty: Uint
+    extra_data: Bytes
+    gas_limit: Uint
+    nonce: Bytes8
+    timestamp: U256
+    alloc: Dict[Address, Dict[str, U256]]
+
+
+def load_genesis_config(genesis_file: str) -> GenesisConfig:
```

Comment:
I think we can simplify this to:

```suggestion
def genesis_configuration() -> GenesisConfiguration:
```
Or maybe:

```suggestion
def genesis_configuration(genesis_path: Optional[str] = None) -> GenesisConfiguration:
```

I don't think configuring the path will be super common, though I guess we should support it so we can sync testnets?

Reply:
Yes the idea is to keep it generic for potentially syncing other chains, though I don't think the rest of the code is **yet** generic to support multiple chains.

Reply:
That makes sense. I guess we'll need to come up with a more generic solution where we can ship testnets as part of the package too, and support loading from an external file.